uaclient/contract.py

```diff
@@ -165,6 +165,17 @@ def detach_machine_from_contract(
         self, machine_token: str, contract_id: str, machine_id: str = None
     ) -> "Dict":
         """Report the attached machine should be detached from the contract."""
+        curr_machine_id = self._get_platform_data(machine_id=None).get(
+            "machineId", ""
+        )
+        past_machine_id = self.cfg.read_cache("machine-id")
```

Comment:
We need a data_paths defintion  "instance-id": DataPath("machine-id", True), otherwise delete_cache will leave files around in /var/lib/ubuntu-advantage/ that we shouldn't.

Also while we are at it, I see we are missing
        "machine-access-cis": DataPath("machine-access-fips.json", True),
and 
        "machine-access-cis": DataPath("machine-access-fips-updates.json", True),
. Can we add check in the features/attached_commands.feature  "detach integration test that asserts /var/lib/ubuntu-advantage is empty after detach

Reply:
@lucasmoura pointed out that we already had a DataPath("machine-id"..) in the module. So disregard this request. I'll open a separate PR.